metaflow/client/core.py

```diff
@@ -398,6 +406,35 @@ def _filtered_children(self, *tags):
     def _url_token(cls):
         return '%ss' % cls._NAME
 
+    def _tag_operations(self, operations):
+        if len(operations) == 0:
+            raise TaggingException("At least one tagging operation must be provided")
+        formatted_operations = []
+        # Perform some sanity checks to report more precisely to the user
+        for o, t in operations:
+            if o not in ['add', 'remove']:
+                raise TaggingException("Invalid tagging operation %s" % o)
+            if o == 'add' and t in self._system_tags:
+                # We don't throw an error but just don't send it; it's a no-op
+                continue
+            if o == 'remove':
+                if t in self._system_tags:
+                    raise TaggingException("Cannot remove system tag *%s*" % t)
+                if t not in self._tags:
+                    raise TaggingException(
+                        "Cannot remove tag *%s* as it does not exist" % t)
```

Comment:
I wonder if removing a non-existent tag should be just a no-op. For instance, currently you can't do ```.replace_tag('production', 'archived')```
without having a separate path for the case if `production` doesn't exist. This is extra tricky, since multiple clients may manipulate tags concurrently